crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -263,7 +263,28 @@ impl UncommittedData {
 }
 
 // TODO: set this via the config
-static MAX_CACHE_SIZE: u64 = 10000;
+static MAX_CACHE_SIZE: u64 = 100000;
+
+// Point items (anything without a version number) can be negatively cached as None
+type PointCacheItem<T> = Option<T>;
+
+// PointCacheItem can only be used for insert-only collections, so a Some entry
+// is always newer than a None entry.
+impl<T: Eq + std::fmt::Debug> IsNewer for PointCacheItem<T> {
+    fn is_newer_than(&self, other: &PointCacheItem<T>) -> bool {
+        match (self, other) {
+            (Some(_), None) => true,
+
+            (Some(a), Some(b)) => {
+                // conflicting inserts should never happen
```

Comment:
is this an invariant violation worth panic-ing for even in release builds?

Reply:
its probably worth it, but i was concerned about possibly causing unnecessary deep-equality checks on transactions/effects